tests/ttnn/unit_tests/operations/test_moreh_nll_loss.py

```diff
@@ -18,38 +18,47 @@
 )
 
 
-def get_torch_tensors(shape):
+def get_torch_tensors(shape, torch_dtype):
     C = shape[1]
     target_shape = shape[:1] + shape[2:]
 
-    cpu_dtype = torch.float32
     cpu_index_dtype = torch.long
 
-    torch_input = torch.rand(shape, dtype=cpu_dtype).requires_grad_()
+    torch_input = torch.rand(shape, dtype=torch_dtype).requires_grad_()
     torch_target = torch.randint(0, C, target_shape, dtype=cpu_index_dtype)
-    torch_weight = torch.rand(C, dtype=cpu_dtype)
-    torch_divisor = torch.tensor([0], dtype=cpu_dtype)
-    torch_output = torch.tensor([0], dtype=cpu_dtype)
+    torch_weight = torch.rand(C, dtype=torch_dtype)
+    torch_divisor = torch.tensor([0], dtype=torch_dtype)
+    torch_output = torch.tensor([0], dtype=torch_dtype)
 
     return torch_input, torch_target, torch_weight, torch_divisor, torch_output
 
 
-def get_tt_tensors(torch_input, torch_target, torch_weight, torch_divisor, torch_output, device):
+def get_tt_tensors(torch_input, torch_target, torch_weight, torch_divisor, torch_output, device, ttnn_dtype):
     npu_index_dtype = ttnn.int32
 
-    tt_input = to_ttnn(torch_input, device=device)
-    tt_target = to_ttnn(torch_target, device=device, dtype=npu_index_dtype)
-    tt_weight = to_ttnn(torch_weight, device=device)
-    tt_divisor = to_ttnn(torch_divisor, device=device)
-    tt_output = to_ttnn(torch_output, device=device)
+    tt_input = to_ttnn(torch_input, dtype=ttnn_dtype, device=device)
+    tt_target = to_ttnn(torch_target, dtype=npu_index_dtype, device=device)
+    tt_weight = to_ttnn(torch_weight, dtype=ttnn_dtype, device=device)
+    tt_divisor = to_ttnn(torch_divisor, dtype=ttnn_dtype, device=device)
+    tt_output = to_ttnn(torch_output, dtype=ttnn_dtype, device=device)
```

Comment:
I suggest using `ttnn.from_torch`

Reply:
If I change this to `ttnn.from_torch` with `tt_weight` tensor, a FATAL err will be raised. It's because `torch_weight` can be `None`. `to_ttnn` is a wrapper of `from_torch` which can handle this case by returning `None` immediately.

Reply:
Can you please help fire a ticket at me. Thank you!

Reply:
I see, in this case, we can keep using `to_ttnn`.
Or: if else if tt_weight.

Reply:
Sure. Here the issue is: https://github.com/tenstorrent/tt-metal/issues/14458